sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ReserveOfferRecommendation.java

```diff
@@ -16,11 +14,16 @@
     private final Operation operation;
 
     public ReserveOfferRecommendation(Offer offer, Resource resource) {
+        if (resource.hasDisk()) {
+            DiskInfo diskInfo = resource.getDisk().toBuilder()
```

Comment:
Needs a comment explaining why this is needed (also, why does in vary from prior `getReservedResource()`)?